src/main/java/uk/gov/digital/ho/hocs/audit/export/CustomExportService.java

```diff
@@ -92,10 +91,13 @@ public void customExport(HttpServletResponse response, String code, boolean conv
                                 printer.printRecord(converted);
                                 outputWriter.flush();
                             } catch (IOException e) {
-                                log.error("Unable to parse record for custom report, reason: {}, event: {}", e.getMessage(), value(LogEvent.EVENT, CSV_EXPORT_FAILURE));
+                                throw new RuntimeException();
```

Comment:
RuntimeException is an _unchecked_ exception and is the wrong use for this kind of issue. Runtime exceptions are when you *can't* recover. 

This should be replaced with our own simple ExportException (extends Exception) that is created with the original stack trace. We should catch that and log..

Reply:
Because it's in a lambda we cant throw a checked exception - and it sounds like extending RuntimeException and throwing that wouldn't be the right solution either. 

I also considered using [takeWhile](https://docs.oracle.com/javase/9/docs/api/java/util/stream/Stream.html#takeWhile-java.util.function.Predicate-) before the forEachOrdered() and having it depend on a boolean which is set to false in the event of an exception to break off early, it's a bit of a misuse of takeWhile but would serve the purpose here. Other than that I think we'd need to convert to a non-stream implementation which don't seem to be really designed for breaking out of.

Reply:
Something like this 
`MutableBoolean ongoing = MutableBoolean.of(true);
someobjects.stream()...takeWhile(t -> ongoing.value()).forEach(t -> {
    // doing something.
    if (...) { // want to break;
        ongoing.setFalse();
    }
});`

Reply:
Sounds like using a regular loop would be better